src/main/java/org/plumelib/javadoc/RequireJavadoc.java

```diff
@@ -289,6 +320,158 @@ boolean shouldExclude(Path path) {
+   */
+  boolean isTrivialGetterOrSetter(MethodDeclaration md) {
+    String methodName = md.getNameAsString();
+    if (methodName.startsWith("get")) {
+      String propertyName = propertyName(md, true);
+      if (propertyName == null) {
+        return false;
+      }
+      Statement statement = getOnlyStatement(md);
+      if (!(statement instanceof ReturnStmt)) {
+        return false;
+      }
+      Optional<Expression> oReturnExpr = ((ReturnStmt) statement).getExpression();
+      if (!oReturnExpr.isPresent()) {
+        return false;
+      }
+      Expression returnExpr = oReturnExpr.get();
+      String returnName;
+      if (returnExpr instanceof NameExpr) {
+        returnName = ((NameExpr) returnExpr).getNameAsString();
+      } else if (returnExpr instanceof FieldAccessExpr) {
+        FieldAccessExpr fa = (FieldAccessExpr) returnExpr;
+        Expression receiver = fa.getScope();
+        if (!(receiver instanceof ThisExpr)) {
+          return false;
+        }
+        returnName = fa.getNameAsString();
+      } else {
+        return false;
+      }
+      if (!returnName.equals(propertyName)) {
+        return false;
+      }
+      return true;
+    } else if (methodName.startsWith("set")) {
+      String propertyName = propertyName(md, false);
+      if (propertyName == null) {
+        return false;
+      }
+      Statement statement = getOnlyStatement(md);
+      if (!(statement instanceof ExpressionStmt)) {
+        return false;
+      }
+      Expression expr = ((ExpressionStmt) statement).getExpression();
+      if (!(expr instanceof AssignExpr)) {
+        return false;
+      }
+      AssignExpr assignExpr = (AssignExpr) expr;
+      Expression target = assignExpr.getTarget();
+      AssignExpr.Operator op = assignExpr.getOperator();
+      Expression value = assignExpr.getValue();
+      if (!(target instanceof FieldAccessExpr)) {
+        return false;
+      }
+      FieldAccessExpr fa = (FieldAccessExpr) target;
+      Expression receiver = fa.getScope();
+      if (!(receiver instanceof ThisExpr)) {
+        return false;
+      }
+      if (!fa.getNameAsString().equals(propertyName)) {
+        return false;
+      }
+      if (op != AssignExpr.Operator.ASSIGN) {
+        return false;
+      }
+      if (!(value instanceof NameExpr
+          && ((NameExpr) value).getNameAsString().equals(propertyName))) {
+        return false;
+      }
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Returns the lower-cased name of the property, if the method is a getter or setter. Otherwise
+   * returns null.
+   *
+   * @param md the method to test
+   * @param isGetter true if testing for a getter, false if testing for a setter
+   * @return the lower-cased names of the property, or null
+   */
+  @Nullable String propertyName(MethodDeclaration md, boolean isGetter) {
+    String methodName = md.getNameAsString();
+    if (!methodName.startsWith(isGetter ? "get" : "set")) {
```

Comment:
That is handled by the `if (upperCaseProperty.length() == 0) {` test a few lines below.

There are no `IndexOutOfBounds` exceptions in this code (at least for arrays and Strings) because each commit is [formally verified](https://github.com/plume-lib/require-javadoc/blob/master/build.gradle#L67) to [use correct indexing](https://checkerframework.org/manual/#index-checker).

I have added a test case, though; thanks for pointing out that this corner case was not tested.